src/main/java/com/iridium/iridiumskyblock/commands/CommandManager.java

```diff
@@ -233,16 +196,40 @@ public boolean onCommand(@NotNull CommandSender commandSender, org.bukkit.comman
 
         // Let the sub-command handle the tab completion
         for (Command command : commands) {
-            if (command.aliases.contains(args[0]) && (
-                    commandSender.hasPermission(command.permission) || command.permission.equalsIgnoreCase("")
-                            || command.permission.equalsIgnoreCase("iridiumskyblock."))) {
-                return command.onTabComplete(commandSender, cmd, label, args);
+            if (command.aliases.contains(args[0])) {
+                Command executingCommand = findExecutingCommand(command, args);
+                if (hasPermissions(commandSender, executingCommand)) {
+                    List<String> tabCompletion = new ArrayList<>(executingCommand.onTabComplete(commandSender, cmd, label, args));
+                    tabCompletion.addAll(executingCommand.getChildNames());
+                    return tabCompletion;
+                }
             }
         }
 
-        // We currently don't want to tab-completion here
         // Return a new List so it isn't a list of online players
         return Collections.emptyList();
     }
 
+    private boolean hasPermissions(@NotNull CommandSender commandSender, Command command) {
+        return commandSender.hasPermission(command.permission)
+            || command.permission.equalsIgnoreCase("")
+            || command.permission.equalsIgnoreCase("iridiumskyblock.");
+    }
+
+    private Command findExecutingCommand(Command baseCommand, String[] args) {
+        Command executingCommand = baseCommand;
+        int currentArgument = 1;
+
+        // Check for each argument if it's a child of the previous command
+        while (args.length > currentArgument) {
```

Comment:
Cant this just be a for loop? or am I being dum?

Reply:
Well, yeah